build/ci.go

```diff
@@ -0,0 +1,61 @@
+// +build none
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+)
+
+func main() {
+
+	log.SetFlags(log.Lshortfile)
+
+	if _, err := os.Stat(filepath.Join("build", "ci.go")); os.IsNotExist(err) {
+		log.Fatal("should run build from root dir")
+	}
+	if len(os.Args) < 2 {
+		log.Fatal("cmd required, eg: install")
+	}
+	switch os.Args[1] {
+	case "install":
+		install()
+	default:
+		log.Fatal("cmd not found", os.Args[1])
+	}
+}
+
+func install() {
```

Comment:
Could you explain this a bit? I don't understand why you run `go install` on every package.

Reply:
that is to avoid issues with weird compilation issues like missing packages / deps.

Reply:
I'm not sure I understand. Is there a particular scenario this would fix?

Reply:
Sure, it will avoid several issues with c modules or cgo issues. The Dockerfile was broken, I first tried to update to last go13 and latest Apline version. I spend a whole afternoon working to fix issues, and mostly related to eg: "could not compile crypto" or "package crypto not found" or missing c libs related to the wasm, or gcc missing, or g++ missing, or other deps that could not be found or compiled. So I added the ci.go into it to avoid the issues. Besides, it will alow people on other distros (eg: windows) to compile it properly without having to dive into go install issues.

Reply:
I'm still not sure how this would fix that. Running `go install` on every package should not be required to build this. Can you share any further details about the errors you ran into? Perhaps you can open an issue?

Reply:
well when you do go install ./... you are doing the same thing but you are mode dependent on the env configurations. When you use go to build go, you can then be sure about the packages are being build the same independently of the OS. In any case, being explicit on the packages and have output on the console is always useful for CI multi cross package builds on go. 

I can share the issues with it, perhaps to try to fix the Alpine build in the future.

Reply:
I've raised https://github.com/ChainSafe/gossamer/issues/424
I would still like to keep the ci.go build because I am not sure if Windows compilation actually works. Please let me know if you want me to ditch it so I can do it and go ahead with the issue.
Cheers.

Reply:
Check out my comment on #424

Reply:
Just replied there, I really would like to have this in asap because its kind of blocking my local build as the `make gossamer` and `make start` are currently not really nice to use :)
Let me know what is your decision, I can remove the go building go if this is what you want.
Cheers.

Reply:
i think I'm starting to see the appeal of this. Certainly as we move to more complex testing configurations this will be useful.

Thanks for your patience, apologies for how this has been dragged out.

Reply:
no worries, its my pleasure to help :)
--
I did the requested changes. Would you like to create a new issue for adding tests into ci.go in a near future ?

Reply:
Yeah that would be great!